docs/design/topics.md

```diff
@@ -0,0 +1,168 @@
+# Topics
+
+Authors: [Philip Metzger](mailto:philipmetzger@bluewin.ch), [Noah Mayr](mailto:dev@noahmayr.com)
+ [Anton Bulakh](mailto:him@necaq.ua)
+
+## Summary
+
+Introduce Topics as a truly Jujutsu native way for topological branches, which 
+also replace the current bookmark concept for Git interop. As they have been
+documented to be confusing users coming from Git. They also supersede the 
+`[experimental-advance-branches]` config for those who currently use it, as 
+such a behavior will be built-in for Topics.
+
+Topics have been discussed heavily since their appearance in 
+[this Discussion][gh-discuss]. As Noah, Anton and I had a long 
+[Discord discussion][dc-thread] about them, which then also poured into the 
+[Topic issue][issue].
+
+## Prior work 
+
+Currently there only is Mercurial which has a implementation of 
+[Topics][hg-topics]. There also is the [Topic feature][gerrit-topics] in Gerrit,
+which groups commits with a single identifier.
+
+
+## Goals and non-goals
+
+### Goals
+
+The goals for this Project are small, see below.
+
+* Introduce the concept of native topological branches for Jujutsu.
+* Simplify Git interop by reducing the burden on Bookmarks.
+* Add Change metadata as a storage concept.
+* Remove the awkward `bookmark` to Git `branch` mapping.
+
+### Non-Goals
+
+* Making bookmarks unnecessary.
+
+## Overview
+
+Until now, Jujutsu had no native set of topological branches, just 
+[Bookmarks][bm] which interact poorly with Git's expectation of branches. 
```

Comment:
I'm not sure if it makes sense to refer to "Git's expectation" separately from the expectations of Git _users_, but if we do want to call out poor interaction between bookmarks and branches, then I think we should be more specific about where the tooling actually diverges in behavior. Git has a concept of a "current branch" that `jj` completely lacks, and it auto-advances the current branch, whereas `jj` doesn't auto-advance bookmarks except using the experimental feature that this design hopes to deprecate. 

```suggestion
[Bookmarks][bm] which interact poorly with Git's expectation of a "current branch" that is advanced automatically as commits are added. 
```